metriq_gym/run.py

```diff
@@ -89,11 +95,60 @@ def poll_job(args: argparse.Namespace, job_manager: JobManager) -> None:
     ]
     if all(task.status() == JobStatus.COMPLETED for task in quantum_job):
         result_data: list[ResultData] = [task.result().data for task in quantum_job]
-        print(handler.poll_handler(job_data, result_data))
+        results: BenchmarkResult = handler.poll_handler(job_data, result_data)
+        print(results)
+        if is_upload:
+            upload_job(args, job_type, job_data, results, platforms[metriq_job.device_name.lower()])
     else:
         logger.info("Job is not yet completed. Please try again later.")
 
 
+def upload_job(args: argparse.Namespace, job_type: JobType, job_data: BenchmarkData, result_data: BenchmarkResult, platform: int):
+    client = MetriqClient(args.token)
+    task = 0
+    method = 0
+    if job_type == JobType.QUANTUM_VOLUME:
+        task = 235 #Quantum Volume task ID
+        method = 144 #Heavy output generation task ID
+    elif job_type == JobType.BSEQ:
+        task = 236 #BSEQ task ID
+        method = 426 #BSEQ method ID
+    else:
+        raise Exception("You're trying to upload an unrecognized job type!")
+    client.submission_add_task(args.submission_id, task)
+    client.submission_add_method(args.submission_id, method)
+    client.submission_add_platform(args.submission_id, platform)
+    result_create_request = ResultCreateRequest(
+        task = str(task),
+        method = str(method),
+        platform = str(platform),
+        isHigherBetter = str(True),
+        metricName = "",
+        metricValue = str(0),
+        evaluatedAt = date.today().strftime("%Y-%m-%d"),
```

Comment:
For the `evaluatedAt`, do we want to include today's date or the timestamp of the associated metriq job?

Reply:
We'd prefer the time the job was run, in the ideal. The problem if we pull it from the providers' job types is that this information won't necessarily have a consistent property name. If we add it to the `BenchmarkData` or `BenchmarkResult`, great, but the polling method shouldn't have to concern itself with provider-specific logic.

Reply:
> the time the job was run
We could have a general way to get when the job has been executed cross-platform, but for now I would stick with the dispatch_time timestamp that we use in list-jobs